src/hooks/useFetchTeams.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useReducer} from 'react';
+import {teamsReducer} from 'reducers/TeamsReducer';
+import {TeamsState} from 'types';
+import {getTeams as fetchTeams} from '../api';
+
+const useFetchTeams = () => {
```

Comment:
custom hooks are useful for reusing the logic in different parts of the app. So we can separate the responsibilities and keep the components more readable and cleaner